### PR TITLE
Order by date / Show filename if no title

### PR DESCRIPTION
**Pull request by geramirez**
Solves
- [show filename if title not present](https://github.com/18F/foia-hub/issues/637)
- [implement ability to sort documents by date](https://github.com/18F/foia-hub/issues/634)
- cleans up GET query strings in search

### CODE REVIEW

File: docusearch/templates/docusearch/detail.html [hunk 1]
```diff
@@ -6,16 +6,22 @@
 <div class="container">
     <section class="content">
     <h1><em>Document</em></h1>
+    {% if document.title %}
```
Comment (khandelwal) [inline docusearch/templates/docusearch/detail.html:10]: Indentation will make this easier to read.

File: docusearch/templates/search/search.html [hunk 2]
```diff
@@ -24,13 +24,31 @@
 
         <div>
         <dl>
-        <h3><a href="?q={{request.GET.q}}&amp;order_by=date_released">Order By</a> </h3>
+        <dd>
+            {% if request.GET.order_by == 'date_released' %}
+            <a href="{{generate_get_req([
```
Comment (khandelwal) [inline docusearch/templates/search/search.html]: This is a lot of logic in the templates. Should we move the generate of these links to the view? That way the code's testable.

**Comment by khandelwal** [comment]
There's a lot of logic in the templates here, that I think needs to be moved into the view.

**Comment by geramirez** [comment]
@khandelwal K, I'll work on cleaning it up.

File: docusearch/templates/search/search.html [hunk 3]
```diff
@@ -24,13 +24,24 @@
...
+            {% else %}
+                <a href="?{{cleaned_query}}&amp;order_by=date_released">Order By Date Released</a>
+            {% endif %}
+        </dd>
         {% if facets.fields.foia_agency %}
             <dt>Agencies</dt>
 
             {% for agency in facets.fields.foia_agency %}
                 <dd>
-                    <a href="?q={{request.GET.q}}&amp;order_by={{request.GET.order_by}}&amp;selected_facets=foia_agency_exact:{{agency.0|urlencode}}">{{agency.0|upper}}</a> ({{ agency.1 }})
```
Comment (ramirezg) [inline docusearch/templates/search/search.html]: Prevents duplicate facets

**Comment by geramirez** [comment]
Okay, this should be ready for review now. @khandelwal

File: docusearch/templates/search/search.html [hunk 4]
```diff
@@ -24,13 +24,24 @@
 
         <div>
         <dl>
-        <h3><a href="?q={{request.GET.q}}&amp;order_by=date_released">Order By</a> </h3>
+        <dd>
+            {% if request.GET.order_by == 'date_released' %}
+                <a href="?{{cleaned_query}}&amp;order_by=relevance ">Order By Date Relevance</a>
```
Comment (khandelwal) [inline docusearch/templates/search/search.html:29]: Calling it cleaned query here isn't useful. What's it a link to? Changing the name will help this be self-documenting. I don't think is this a blocker for this pull request - but can we change it subsequently?

**Pull request merged**